connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -69,8 +69,8 @@
     private Converter internalKeyConverter;
     private Converter internalValueConverter;
     private OffsetBackingStore offsetBackingStore;
-    private HashMap<String, Connector> connectors = new HashMap<>();
-    private HashMap<ConnectorTaskId, WorkerTask> tasks = new HashMap<>();
+    private HashMap<String, ConnectorProfile> connectors = new HashMap<>();
+    private HashMap<ConnectorTaskId, TaskProfile> tasks = new HashMap<>();
```

Comment:
Use ConnectorProfile and TaskProfile to store the config of connector and tasks, use the information to restart connector and task;